kuksa-sdk/src/main/kotlin/org/eclipse/kuksa/DataBrokerApiInteraction.kt

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.eclipse.kuksa
+
+import io.grpc.Context
+import io.grpc.ManagedChannel
+import io.grpc.StatusRuntimeException
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.eclipse.kuksa.proto.v1.KuksaValV1
+import org.eclipse.kuksa.proto.v1.Types
+import org.eclipse.kuksa.proto.v1.Types.Field
+import org.eclipse.kuksa.proto.v1.VALGrpc
+import org.eclipse.kuksa.subscription.Subscription
+
+/**
+ * Encapsulates the interactions with the DataBroker using gRPC.
+ */
+internal class DataBrokerApiInteraction(
+    private val managedChannel: ManagedChannel,
+    private val defaultDispatcher: CoroutineDispatcher = Dispatchers.Default,
+) {
+    /**
+     * Sends a request to the DataBroker to respond with the specified [vssPath] and [fields] values.
+     *
+     * Throws a [DataBrokerException] in case the connection to the DataBroker is no longer active
+     */
+    suspend fun fetchProperty(
```

Comment:
Either just "fetch" or rename subscribe to "subscribeProperty".  Why are we not using the "Property" model here?

Reply:
At such a low level - if not absolutely necessary - I want to be independent of any and all constraints a model gives me. e.g. the Property enforces me to use a list of Fields (which is fine for most of the methods but not for subscribe due to how subscriptions are stored inside the map)

The only exception here is obviously the return-type "Subscription" which I return because i need to somehow "cache" the cancellableContext for unsubscribing and there being a hard dependency on the internal "inner class Observer" anyway. But Subscription is internal and therefore not for use inside the pubic API. 
I don't want to make use of a Pair or a Triple to transport such Data outside of the method.

Reply:
Renamed fetchProperty and updateProperty to fetch and update